// @aws-sdk/client-redshift/dist/es/commands/BatchDeleteClusterSnapshotsCommand.js
import { __extends } from "tslib";
import { BatchDeleteClusterSnapshotsRequest, BatchDeleteClusterSnapshotsResult } from "../models/models_0";
import { deserializeAws_queryBatchDeleteClusterSnapshotsCommand, serializeAws_queryBatchDeleteClusterSnapshotsCommand, } from "../protocols/Aws_query";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
/**
 * <p>Deletes a set of cluster snapshots.</p>
 */
var BatchDeleteClusterSnapshotsCommand = /** @class */ (function (_super) {
    __extends(BatchDeleteClusterSnapshotsCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function BatchDeleteClusterSnapshotsCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        _this.resolved = false;
        return _this;
        // End section: command_constructor
    }
    /**
     * @internal
     */
    BatchDeleteClusterSnapshotsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        if (!this.resolved) {
            this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
            this.resolved = true;
        }
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftClient";
        var commandName = "BatchDeleteClusterSnapshotsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: BatchDeleteClusterSnapshotsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: BatchDeleteClusterSnapshotsResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    BatchDeleteClusterSnapshotsCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryBatchDeleteClusterSnapshotsCommand(input, context);
    };
    BatchDeleteClusterSnapshotsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryBatchDeleteClusterSnapshotsCommand(output, context);
    };
    return BatchDeleteClusterSnapshotsCommand;
}($Command));
export { BatchDeleteClusterSnapshotsCommand };
