// @aws-sdk/client-redshift/dist/es/commands/CancelResizeCommand.js
import { __extends } from "tslib";
import { CancelResizeMessage, ResizeProgressMessage } from "../models/models_0";
import { deserializeAws_queryCancelResizeCommand, serializeAws_queryCancelResizeCommand } from "../protocols/Aws_query";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
/**
 * <p>Cancels a resize operation for a cluster.</p>
 */
var CancelResizeCommand = /** @class */ (function (_super) {
    __extends(CancelResizeCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function CancelResizeCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        _this.resolved = false;
        return _this;
        // End section: command_constructor
    }
    /**
     * @internal
     */
    CancelResizeCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        if (!this.resolved) {
            this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
            this.resolved = true;
        }
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftClient";
        var commandName = "CancelResizeCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CancelResizeMessage.filterSensitiveLog,
            outputFilterSensitiveLog: ResizeProgressMessage.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CancelResizeCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryCancelResizeCommand(input, context);
    };
    CancelResizeCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryCancelResizeCommand(output, context);
    };
    return CancelResizeCommand;
}($Command));
export { CancelResizeCommand };
