// @aws-sdk/client-redshift/dist/es/commands/DescribeHsmConfigurationsCommand.js
import { __extends } from "tslib";
import { DescribeHsmConfigurationsMessage, HsmConfigurationMessage } from "../models/models_0";
import { deserializeAws_queryDescribeHsmConfigurationsCommand, serializeAws_queryDescribeHsmConfigurationsCommand, } from "../protocols/Aws_query";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
/**
 * <p>Returns information about the specified Amazon Redshift HSM configuration. If no
 *             configuration ID is specified, returns information about all the HSM configurations
 *             owned by your AWS customer account.</p>
 *         <p>If you specify both tag keys and tag values in the same request, Amazon Redshift returns
 *             all HSM connections that match any combination of the specified keys and values. For
 *             example, if you have <code>owner</code> and <code>environment</code> for tag keys, and
 *                 <code>admin</code> and <code>test</code> for tag values, all HSM connections that
 *             have any combination of those values are returned.</p>
 *         <p>If both tag keys and values are omitted from the request, HSM connections are
 *             returned regardless of whether they have tag keys or values associated with
 *             them.</p>
 */
var DescribeHsmConfigurationsCommand = /** @class */ (function (_super) {
    __extends(DescribeHsmConfigurationsCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function DescribeHsmConfigurationsCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        _this.resolved = false;
        return _this;
        // End section: command_constructor
    }
    /**
     * @internal
     */
    DescribeHsmConfigurationsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        if (!this.resolved) {
            this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
            this.resolved = true;
        }
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftClient";
        var commandName = "DescribeHsmConfigurationsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeHsmConfigurationsMessage.filterSensitiveLog,
            outputFilterSensitiveLog: HsmConfigurationMessage.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeHsmConfigurationsCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDescribeHsmConfigurationsCommand(input, context);
    };
    DescribeHsmConfigurationsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDescribeHsmConfigurationsCommand(output, context);
    };
    return DescribeHsmConfigurationsCommand;
}($Command));
export { DescribeHsmConfigurationsCommand };
