// @aws-sdk/client-redshift/dist/es/commands/EnableLoggingCommand.js
import { __extends } from "tslib";
import { EnableLoggingMessage, LoggingStatus } from "../models/models_0";
import { deserializeAws_queryEnableLoggingCommand, serializeAws_queryEnableLoggingCommand, } from "../protocols/Aws_query";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
/**
 * <p>Starts logging information, such as queries and connection attempts, for the
 *             specified Amazon Redshift cluster.</p>
 */
var EnableLoggingCommand = /** @class */ (function (_super) {
    __extends(EnableLoggingCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function EnableLoggingCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        _this.resolved = false;
        return _this;
        // End section: command_constructor
    }
    /**
     * @internal
     */
    EnableLoggingCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        if (!this.resolved) {
            this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
            this.resolved = true;
        }
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftClient";
        var commandName = "EnableLoggingCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: EnableLoggingMessage.filterSensitiveLog,
            outputFilterSensitiveLog: LoggingStatus.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    EnableLoggingCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryEnableLoggingCommand(input, context);
    };
    EnableLoggingCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryEnableLoggingCommand(output, context);
    };
    return EnableLoggingCommand;
}($Command));
export { EnableLoggingCommand };
