// @aws-sdk/client-redshift/dist/es/commands/ModifyClusterSubnetGroupCommand.js
import { __extends } from "tslib";
import { ModifyClusterSubnetGroupMessage, ModifyClusterSubnetGroupResult } from "../models/models_1";
import { deserializeAws_queryModifyClusterSubnetGroupCommand, serializeAws_queryModifyClusterSubnetGroupCommand, } from "../protocols/Aws_query";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
/**
 * <p>Modifies a cluster subnet group to include the specified list of VPC subnets. The
 *             operation replaces the existing list of subnets with the new list of subnets.</p>
 */
var ModifyClusterSubnetGroupCommand = /** @class */ (function (_super) {
    __extends(ModifyClusterSubnetGroupCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function ModifyClusterSubnetGroupCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        _this.resolved = false;
        return _this;
        // End section: command_constructor
    }
    /**
     * @internal
     */
    ModifyClusterSubnetGroupCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        if (!this.resolved) {
            this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
            this.resolved = true;
        }
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftClient";
        var commandName = "ModifyClusterSubnetGroupCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ModifyClusterSubnetGroupMessage.filterSensitiveLog,
            outputFilterSensitiveLog: ModifyClusterSubnetGroupResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ModifyClusterSubnetGroupCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryModifyClusterSubnetGroupCommand(input, context);
    };
    ModifyClusterSubnetGroupCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryModifyClusterSubnetGroupCommand(output, context);
    };
    return ModifyClusterSubnetGroupCommand;
}($Command));
export { ModifyClusterSubnetGroupCommand };
