// @aws-sdk/client-redshift/dist/es/commands/ModifyEventSubscriptionCommand.js
import { __extends } from "tslib";
import { ModifyEventSubscriptionMessage, ModifyEventSubscriptionResult } from "../models/models_1";
import { deserializeAws_queryModifyEventSubscriptionCommand, serializeAws_queryModifyEventSubscriptionCommand, } from "../protocols/Aws_query";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
/**
 * <p>Modifies an existing Amazon Redshift event notification subscription.</p>
 */
var ModifyEventSubscriptionCommand = /** @class */ (function (_super) {
    __extends(ModifyEventSubscriptionCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function ModifyEventSubscriptionCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        _this.resolved = false;
        return _this;
        // End section: command_constructor
    }
    /**
     * @internal
     */
    ModifyEventSubscriptionCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        if (!this.resolved) {
            this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
            this.resolved = true;
        }
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftClient";
        var commandName = "ModifyEventSubscriptionCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ModifyEventSubscriptionMessage.filterSensitiveLog,
            outputFilterSensitiveLog: ModifyEventSubscriptionResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ModifyEventSubscriptionCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryModifyEventSubscriptionCommand(input, context);
    };
    ModifyEventSubscriptionCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryModifyEventSubscriptionCommand(output, context);
    };
    return ModifyEventSubscriptionCommand;
}($Command));
export { ModifyEventSubscriptionCommand };
