// @aws-sdk/client-redshift/dist/cjs/models/models_1.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RotateEncryptionKeyResult = exports.RotateEncryptionKeyMessage = exports.RevokeSnapshotAccessResult = exports.RevokeSnapshotAccessMessage = exports.RevokeClusterSecurityGroupIngressResult = exports.RevokeClusterSecurityGroupIngressMessage = exports.ResumeClusterResult = exports.RestoreTableFromClusterSnapshotResult = exports.RestoreTableFromClusterSnapshotMessage = exports.RestoreFromClusterSnapshotResult = exports.RestoreFromClusterSnapshotMessage = exports.ResizeClusterResult = exports.ResetClusterParameterGroupMessage = exports.RebootClusterResult = exports.RebootClusterMessage = exports.ReservedNodeQuotaExceededFault = exports.PurchaseReservedNodeOfferingResult = exports.PurchaseReservedNodeOfferingMessage = exports.PauseClusterResult = exports.ModifyUsageLimitMessage = exports.SnapshotScheduleUpdateInProgressFault = exports.ModifySnapshotScheduleMessage = exports.SnapshotCopyDisabledFault = exports.ModifySnapshotCopyRetentionPeriodResult = exports.ModifySnapshotCopyRetentionPeriodMessage = exports.ModifyScheduledActionMessage = exports.ModifyEventSubscriptionResult = exports.ModifyEventSubscriptionMessage = exports.SubnetAlreadyInUse = exports.ModifyClusterSubnetGroupResult = exports.ModifyClusterSubnetGroupMessage = exports.ModifyClusterSnapshotScheduleMessage = exports.ModifyClusterSnapshotResult = exports.ModifyClusterSnapshotMessage = exports.ModifyClusterParameterGroupMessage = exports.ModifyClusterMaintenanceResult = exports.ModifyClusterMaintenanceMessage = exports.ModifyClusterIamRolesResult = exports.ModifyClusterIamRolesMessage = exports.ModifyClusterDbRevisionResult = exports.ModifyClusterDbRevisionMessage = exports.UnsupportedOptionFault = exports.TableLimitExceededFault = exports.ModifyClusterResult = exports.ModifyClusterMessage = exports.InvalidTableRestoreArgumentFault = void 0;
var InvalidTableRestoreArgumentFault;
(function (InvalidTableRestoreArgumentFault) {
    InvalidTableRestoreArgumentFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(InvalidTableRestoreArgumentFault = exports.InvalidTableRestoreArgumentFault || (exports.InvalidTableRestoreArgumentFault = {}));
var ModifyClusterMessage;
(function (ModifyClusterMessage) {
    ModifyClusterMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ModifyClusterMessage = exports.ModifyClusterMessage || (exports.ModifyClusterMessage = {}));
var ModifyClusterResult;
(function (ModifyClusterResult) {
    ModifyClusterResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ModifyClusterResult = exports.ModifyClusterResult || (exports.ModifyClusterResult = {}));
var TableLimitExceededFault;
(function (TableLimitExceededFault) {
    TableLimitExceededFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(TableLimitExceededFault = exports.TableLimitExceededFault || (exports.TableLimitExceededFault = {}));
var UnsupportedOptionFault;
(function (UnsupportedOptionFault) {
    UnsupportedOptionFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UnsupportedOptionFault = exports.UnsupportedOptionFault || (exports.UnsupportedOptionFault = {}));
var ModifyClusterDbRevisionMessage;
(function (ModifyClusterDbRevisionMessage) {
    ModifyClusterDbRevisionMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ModifyClusterDbRevisionMessage = exports.ModifyClusterDbRevisionMessage || (exports.ModifyClusterDbRevisionMessage = {}));
var ModifyClusterDbRevisionResult;
(function (ModifyClusterDbRevisionResult) {
    ModifyClusterDbRevisionResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ModifyClusterDbRevisionResult = exports.ModifyClusterDbRevisionResult || (exports.ModifyClusterDbRevisionResult = {}));
var ModifyClusterIamRolesMessage;
(function (ModifyClusterIamRolesMessage) {
    ModifyClusterIamRolesMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ModifyClusterIamRolesMessage = exports.ModifyClusterIamRolesMessage || (exports.ModifyClusterIamRolesMessage = {}));
var ModifyClusterIamRolesResult;
(function (ModifyClusterIamRolesResult) {
    ModifyClusterIamRolesResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ModifyClusterIamRolesResult = exports.ModifyClusterIamRolesResult || (exports.ModifyClusterIamRolesResult = {}));
var ModifyClusterMaintenanceMessage;
(function (ModifyClusterMaintenanceMessage) {
    ModifyClusterMaintenanceMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ModifyClusterMaintenanceMessage = exports.ModifyClusterMaintenanceMessage || (exports.ModifyClusterMaintenanceMessage = {}));
var ModifyClusterMaintenanceResult;
(function (ModifyClusterMaintenanceResult) {
    ModifyClusterMaintenanceResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ModifyClusterMaintenanceResult = exports.ModifyClusterMaintenanceResult || (exports.ModifyClusterMaintenanceResult = {}));
var ModifyClusterParameterGroupMessage;
(function (ModifyClusterParameterGroupMessage) {
    ModifyClusterParameterGroupMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ModifyClusterParameterGroupMessage = exports.ModifyClusterParameterGroupMessage || (exports.ModifyClusterParameterGroupMessage = {}));
var ModifyClusterSnapshotMessage;
(function (ModifyClusterSnapshotMessage) {
    ModifyClusterSnapshotMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ModifyClusterSnapshotMessage = exports.ModifyClusterSnapshotMessage || (exports.ModifyClusterSnapshotMessage = {}));
var ModifyClusterSnapshotResult;
(function (ModifyClusterSnapshotResult) {
    ModifyClusterSnapshotResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ModifyClusterSnapshotResult = exports.ModifyClusterSnapshotResult || (exports.ModifyClusterSnapshotResult = {}));
var ModifyClusterSnapshotScheduleMessage;
(function (ModifyClusterSnapshotScheduleMessage) {
    ModifyClusterSnapshotScheduleMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ModifyClusterSnapshotScheduleMessage = exports.ModifyClusterSnapshotScheduleMessage || (exports.ModifyClusterSnapshotScheduleMessage = {}));
var ModifyClusterSubnetGroupMessage;
(function (ModifyClusterSubnetGroupMessage) {
    ModifyClusterSubnetGroupMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ModifyClusterSubnetGroupMessage = exports.ModifyClusterSubnetGroupMessage || (exports.ModifyClusterSubnetGroupMessage = {}));
var ModifyClusterSubnetGroupResult;
(function (ModifyClusterSubnetGroupResult) {
    ModifyClusterSubnetGroupResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ModifyClusterSubnetGroupResult = exports.ModifyClusterSubnetGroupResult || (exports.ModifyClusterSubnetGroupResult = {}));
var SubnetAlreadyInUse;
(function (SubnetAlreadyInUse) {
    SubnetAlreadyInUse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(SubnetAlreadyInUse = exports.SubnetAlreadyInUse || (exports.SubnetAlreadyInUse = {}));
var ModifyEventSubscriptionMessage;
(function (ModifyEventSubscriptionMessage) {
    ModifyEventSubscriptionMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ModifyEventSubscriptionMessage = exports.ModifyEventSubscriptionMessage || (exports.ModifyEventSubscriptionMessage = {}));
var ModifyEventSubscriptionResult;
(function (ModifyEventSubscriptionResult) {
    ModifyEventSubscriptionResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ModifyEventSubscriptionResult = exports.ModifyEventSubscriptionResult || (exports.ModifyEventSubscriptionResult = {}));
var ModifyScheduledActionMessage;
(function (ModifyScheduledActionMessage) {
    ModifyScheduledActionMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ModifyScheduledActionMessage = exports.ModifyScheduledActionMessage || (exports.ModifyScheduledActionMessage = {}));
var ModifySnapshotCopyRetentionPeriodMessage;
(function (ModifySnapshotCopyRetentionPeriodMessage) {
    ModifySnapshotCopyRetentionPeriodMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ModifySnapshotCopyRetentionPeriodMessage = exports.ModifySnapshotCopyRetentionPeriodMessage || (exports.ModifySnapshotCopyRetentionPeriodMessage = {}));
var ModifySnapshotCopyRetentionPeriodResult;
(function (ModifySnapshotCopyRetentionPeriodResult) {
    ModifySnapshotCopyRetentionPeriodResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ModifySnapshotCopyRetentionPeriodResult = exports.ModifySnapshotCopyRetentionPeriodResult || (exports.ModifySnapshotCopyRetentionPeriodResult = {}));
var SnapshotCopyDisabledFault;
(function (SnapshotCopyDisabledFault) {
    SnapshotCopyDisabledFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(SnapshotCopyDisabledFault = exports.SnapshotCopyDisabledFault || (exports.SnapshotCopyDisabledFault = {}));
var ModifySnapshotScheduleMessage;
(function (ModifySnapshotScheduleMessage) {
    ModifySnapshotScheduleMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ModifySnapshotScheduleMessage = exports.ModifySnapshotScheduleMessage || (exports.ModifySnapshotScheduleMessage = {}));
var SnapshotScheduleUpdateInProgressFault;
(function (SnapshotScheduleUpdateInProgressFault) {
    SnapshotScheduleUpdateInProgressFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(SnapshotScheduleUpdateInProgressFault = exports.SnapshotScheduleUpdateInProgressFault || (exports.SnapshotScheduleUpdateInProgressFault = {}));
var ModifyUsageLimitMessage;
(function (ModifyUsageLimitMessage) {
    ModifyUsageLimitMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ModifyUsageLimitMessage = exports.ModifyUsageLimitMessage || (exports.ModifyUsageLimitMessage = {}));
var PauseClusterResult;
(function (PauseClusterResult) {
    PauseClusterResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(PauseClusterResult = exports.PauseClusterResult || (exports.PauseClusterResult = {}));
var PurchaseReservedNodeOfferingMessage;
(function (PurchaseReservedNodeOfferingMessage) {
    PurchaseReservedNodeOfferingMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(PurchaseReservedNodeOfferingMessage = exports.PurchaseReservedNodeOfferingMessage || (exports.PurchaseReservedNodeOfferingMessage = {}));
var PurchaseReservedNodeOfferingResult;
(function (PurchaseReservedNodeOfferingResult) {
    PurchaseReservedNodeOfferingResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(PurchaseReservedNodeOfferingResult = exports.PurchaseReservedNodeOfferingResult || (exports.PurchaseReservedNodeOfferingResult = {}));
var ReservedNodeQuotaExceededFault;
(function (ReservedNodeQuotaExceededFault) {
    ReservedNodeQuotaExceededFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ReservedNodeQuotaExceededFault = exports.ReservedNodeQuotaExceededFault || (exports.ReservedNodeQuotaExceededFault = {}));
var RebootClusterMessage;
(function (RebootClusterMessage) {
    RebootClusterMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(RebootClusterMessage = exports.RebootClusterMessage || (exports.RebootClusterMessage = {}));
var RebootClusterResult;
(function (RebootClusterResult) {
    RebootClusterResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(RebootClusterResult = exports.RebootClusterResult || (exports.RebootClusterResult = {}));
var ResetClusterParameterGroupMessage;
(function (ResetClusterParameterGroupMessage) {
    ResetClusterParameterGroupMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ResetClusterParameterGroupMessage = exports.ResetClusterParameterGroupMessage || (exports.ResetClusterParameterGroupMessage = {}));
var ResizeClusterResult;
(function (ResizeClusterResult) {
    ResizeClusterResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ResizeClusterResult = exports.ResizeClusterResult || (exports.ResizeClusterResult = {}));
var RestoreFromClusterSnapshotMessage;
(function (RestoreFromClusterSnapshotMessage) {
    RestoreFromClusterSnapshotMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(RestoreFromClusterSnapshotMessage = exports.RestoreFromClusterSnapshotMessage || (exports.RestoreFromClusterSnapshotMessage = {}));
var RestoreFromClusterSnapshotResult;
(function (RestoreFromClusterSnapshotResult) {
    RestoreFromClusterSnapshotResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(RestoreFromClusterSnapshotResult = exports.RestoreFromClusterSnapshotResult || (exports.RestoreFromClusterSnapshotResult = {}));
var RestoreTableFromClusterSnapshotMessage;
(function (RestoreTableFromClusterSnapshotMessage) {
    RestoreTableFromClusterSnapshotMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(RestoreTableFromClusterSnapshotMessage = exports.RestoreTableFromClusterSnapshotMessage || (exports.RestoreTableFromClusterSnapshotMessage = {}));
var RestoreTableFromClusterSnapshotResult;
(function (RestoreTableFromClusterSnapshotResult) {
    RestoreTableFromClusterSnapshotResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(RestoreTableFromClusterSnapshotResult = exports.RestoreTableFromClusterSnapshotResult || (exports.RestoreTableFromClusterSnapshotResult = {}));
var ResumeClusterResult;
(function (ResumeClusterResult) {
    ResumeClusterResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ResumeClusterResult = exports.ResumeClusterResult || (exports.ResumeClusterResult = {}));
var RevokeClusterSecurityGroupIngressMessage;
(function (RevokeClusterSecurityGroupIngressMessage) {
    RevokeClusterSecurityGroupIngressMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(RevokeClusterSecurityGroupIngressMessage = exports.RevokeClusterSecurityGroupIngressMessage || (exports.RevokeClusterSecurityGroupIngressMessage = {}));
var RevokeClusterSecurityGroupIngressResult;
(function (RevokeClusterSecurityGroupIngressResult) {
    RevokeClusterSecurityGroupIngressResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(RevokeClusterSecurityGroupIngressResult = exports.RevokeClusterSecurityGroupIngressResult || (exports.RevokeClusterSecurityGroupIngressResult = {}));
var RevokeSnapshotAccessMessage;
(function (RevokeSnapshotAccessMessage) {
    RevokeSnapshotAccessMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(RevokeSnapshotAccessMessage = exports.RevokeSnapshotAccessMessage || (exports.RevokeSnapshotAccessMessage = {}));
var RevokeSnapshotAccessResult;
(function (RevokeSnapshotAccessResult) {
    RevokeSnapshotAccessResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(RevokeSnapshotAccessResult = exports.RevokeSnapshotAccessResult || (exports.RevokeSnapshotAccessResult = {}));
var RotateEncryptionKeyMessage;
(function (RotateEncryptionKeyMessage) {
    RotateEncryptionKeyMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(RotateEncryptionKeyMessage = exports.RotateEncryptionKeyMessage || (exports.RotateEncryptionKeyMessage = {}));
var RotateEncryptionKeyResult;
(function (RotateEncryptionKeyResult) {
    RotateEncryptionKeyResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(RotateEncryptionKeyResult = exports.RotateEncryptionKeyResult || (exports.RotateEncryptionKeyResult = {}));
