// @aws-sdk/client-redshift/dist/es/models/models_1.js
import { __assign } from "tslib";
export var InvalidTableRestoreArgumentFault;
(function (InvalidTableRestoreArgumentFault) {
    InvalidTableRestoreArgumentFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InvalidTableRestoreArgumentFault || (InvalidTableRestoreArgumentFault = {}));
export var ModifyClusterMessage;
(function (ModifyClusterMessage) {
    ModifyClusterMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ModifyClusterMessage || (ModifyClusterMessage = {}));
export var ModifyClusterResult;
(function (ModifyClusterResult) {
    ModifyClusterResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ModifyClusterResult || (ModifyClusterResult = {}));
export var TableLimitExceededFault;
(function (TableLimitExceededFault) {
    TableLimitExceededFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(TableLimitExceededFault || (TableLimitExceededFault = {}));
export var UnsupportedOptionFault;
(function (UnsupportedOptionFault) {
    UnsupportedOptionFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UnsupportedOptionFault || (UnsupportedOptionFault = {}));
export var ModifyClusterDbRevisionMessage;
(function (ModifyClusterDbRevisionMessage) {
    ModifyClusterDbRevisionMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ModifyClusterDbRevisionMessage || (ModifyClusterDbRevisionMessage = {}));
export var ModifyClusterDbRevisionResult;
(function (ModifyClusterDbRevisionResult) {
    ModifyClusterDbRevisionResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ModifyClusterDbRevisionResult || (ModifyClusterDbRevisionResult = {}));
export var ModifyClusterIamRolesMessage;
(function (ModifyClusterIamRolesMessage) {
    ModifyClusterIamRolesMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ModifyClusterIamRolesMessage || (ModifyClusterIamRolesMessage = {}));
export var ModifyClusterIamRolesResult;
(function (ModifyClusterIamRolesResult) {
    ModifyClusterIamRolesResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ModifyClusterIamRolesResult || (ModifyClusterIamRolesResult = {}));
export var ModifyClusterMaintenanceMessage;
(function (ModifyClusterMaintenanceMessage) {
    ModifyClusterMaintenanceMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ModifyClusterMaintenanceMessage || (ModifyClusterMaintenanceMessage = {}));
export var ModifyClusterMaintenanceResult;
(function (ModifyClusterMaintenanceResult) {
    ModifyClusterMaintenanceResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ModifyClusterMaintenanceResult || (ModifyClusterMaintenanceResult = {}));
export var ModifyClusterParameterGroupMessage;
(function (ModifyClusterParameterGroupMessage) {
    ModifyClusterParameterGroupMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ModifyClusterParameterGroupMessage || (ModifyClusterParameterGroupMessage = {}));
export var ModifyClusterSnapshotMessage;
(function (ModifyClusterSnapshotMessage) {
    ModifyClusterSnapshotMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ModifyClusterSnapshotMessage || (ModifyClusterSnapshotMessage = {}));
export var ModifyClusterSnapshotResult;
(function (ModifyClusterSnapshotResult) {
    ModifyClusterSnapshotResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ModifyClusterSnapshotResult || (ModifyClusterSnapshotResult = {}));
export var ModifyClusterSnapshotScheduleMessage;
(function (ModifyClusterSnapshotScheduleMessage) {
    ModifyClusterSnapshotScheduleMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ModifyClusterSnapshotScheduleMessage || (ModifyClusterSnapshotScheduleMessage = {}));
export var ModifyClusterSubnetGroupMessage;
(function (ModifyClusterSubnetGroupMessage) {
    ModifyClusterSubnetGroupMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ModifyClusterSubnetGroupMessage || (ModifyClusterSubnetGroupMessage = {}));
export var ModifyClusterSubnetGroupResult;
(function (ModifyClusterSubnetGroupResult) {
    ModifyClusterSubnetGroupResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ModifyClusterSubnetGroupResult || (ModifyClusterSubnetGroupResult = {}));
export var SubnetAlreadyInUse;
(function (SubnetAlreadyInUse) {
    SubnetAlreadyInUse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SubnetAlreadyInUse || (SubnetAlreadyInUse = {}));
export var ModifyEventSubscriptionMessage;
(function (ModifyEventSubscriptionMessage) {
    ModifyEventSubscriptionMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ModifyEventSubscriptionMessage || (ModifyEventSubscriptionMessage = {}));
export var ModifyEventSubscriptionResult;
(function (ModifyEventSubscriptionResult) {
    ModifyEventSubscriptionResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ModifyEventSubscriptionResult || (ModifyEventSubscriptionResult = {}));
export var ModifyScheduledActionMessage;
(function (ModifyScheduledActionMessage) {
    ModifyScheduledActionMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ModifyScheduledActionMessage || (ModifyScheduledActionMessage = {}));
export var ModifySnapshotCopyRetentionPeriodMessage;
(function (ModifySnapshotCopyRetentionPeriodMessage) {
    ModifySnapshotCopyRetentionPeriodMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ModifySnapshotCopyRetentionPeriodMessage || (ModifySnapshotCopyRetentionPeriodMessage = {}));
export var ModifySnapshotCopyRetentionPeriodResult;
(function (ModifySnapshotCopyRetentionPeriodResult) {
    ModifySnapshotCopyRetentionPeriodResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ModifySnapshotCopyRetentionPeriodResult || (ModifySnapshotCopyRetentionPeriodResult = {}));
export var SnapshotCopyDisabledFault;
(function (SnapshotCopyDisabledFault) {
    SnapshotCopyDisabledFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SnapshotCopyDisabledFault || (SnapshotCopyDisabledFault = {}));
export var ModifySnapshotScheduleMessage;
(function (ModifySnapshotScheduleMessage) {
    ModifySnapshotScheduleMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ModifySnapshotScheduleMessage || (ModifySnapshotScheduleMessage = {}));
export var SnapshotScheduleUpdateInProgressFault;
(function (SnapshotScheduleUpdateInProgressFault) {
    SnapshotScheduleUpdateInProgressFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SnapshotScheduleUpdateInProgressFault || (SnapshotScheduleUpdateInProgressFault = {}));
export var ModifyUsageLimitMessage;
(function (ModifyUsageLimitMessage) {
    ModifyUsageLimitMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ModifyUsageLimitMessage || (ModifyUsageLimitMessage = {}));
export var PauseClusterResult;
(function (PauseClusterResult) {
    PauseClusterResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PauseClusterResult || (PauseClusterResult = {}));
export var PurchaseReservedNodeOfferingMessage;
(function (PurchaseReservedNodeOfferingMessage) {
    PurchaseReservedNodeOfferingMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PurchaseReservedNodeOfferingMessage || (PurchaseReservedNodeOfferingMessage = {}));
export var PurchaseReservedNodeOfferingResult;
(function (PurchaseReservedNodeOfferingResult) {
    PurchaseReservedNodeOfferingResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PurchaseReservedNodeOfferingResult || (PurchaseReservedNodeOfferingResult = {}));
export var ReservedNodeQuotaExceededFault;
(function (ReservedNodeQuotaExceededFault) {
    ReservedNodeQuotaExceededFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ReservedNodeQuotaExceededFault || (ReservedNodeQuotaExceededFault = {}));
export var RebootClusterMessage;
(function (RebootClusterMessage) {
    RebootClusterMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RebootClusterMessage || (RebootClusterMessage = {}));
export var RebootClusterResult;
(function (RebootClusterResult) {
    RebootClusterResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RebootClusterResult || (RebootClusterResult = {}));
export var ResetClusterParameterGroupMessage;
(function (ResetClusterParameterGroupMessage) {
    ResetClusterParameterGroupMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ResetClusterParameterGroupMessage || (ResetClusterParameterGroupMessage = {}));
export var ResizeClusterResult;
(function (ResizeClusterResult) {
    ResizeClusterResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ResizeClusterResult || (ResizeClusterResult = {}));
export var RestoreFromClusterSnapshotMessage;
(function (RestoreFromClusterSnapshotMessage) {
    RestoreFromClusterSnapshotMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RestoreFromClusterSnapshotMessage || (RestoreFromClusterSnapshotMessage = {}));
export var RestoreFromClusterSnapshotResult;
(function (RestoreFromClusterSnapshotResult) {
    RestoreFromClusterSnapshotResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RestoreFromClusterSnapshotResult || (RestoreFromClusterSnapshotResult = {}));
export var RestoreTableFromClusterSnapshotMessage;
(function (RestoreTableFromClusterSnapshotMessage) {
    RestoreTableFromClusterSnapshotMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RestoreTableFromClusterSnapshotMessage || (RestoreTableFromClusterSnapshotMessage = {}));
export var RestoreTableFromClusterSnapshotResult;
(function (RestoreTableFromClusterSnapshotResult) {
    RestoreTableFromClusterSnapshotResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RestoreTableFromClusterSnapshotResult || (RestoreTableFromClusterSnapshotResult = {}));
export var ResumeClusterResult;
(function (ResumeClusterResult) {
    ResumeClusterResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ResumeClusterResult || (ResumeClusterResult = {}));
export var RevokeClusterSecurityGroupIngressMessage;
(function (RevokeClusterSecurityGroupIngressMessage) {
    RevokeClusterSecurityGroupIngressMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RevokeClusterSecurityGroupIngressMessage || (RevokeClusterSecurityGroupIngressMessage = {}));
export var RevokeClusterSecurityGroupIngressResult;
(function (RevokeClusterSecurityGroupIngressResult) {
    RevokeClusterSecurityGroupIngressResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RevokeClusterSecurityGroupIngressResult || (RevokeClusterSecurityGroupIngressResult = {}));
export var RevokeSnapshotAccessMessage;
(function (RevokeSnapshotAccessMessage) {
    RevokeSnapshotAccessMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RevokeSnapshotAccessMessage || (RevokeSnapshotAccessMessage = {}));
export var RevokeSnapshotAccessResult;
(function (RevokeSnapshotAccessResult) {
    RevokeSnapshotAccessResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RevokeSnapshotAccessResult || (RevokeSnapshotAccessResult = {}));
export var RotateEncryptionKeyMessage;
(function (RotateEncryptionKeyMessage) {
    RotateEncryptionKeyMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RotateEncryptionKeyMessage || (RotateEncryptionKeyMessage = {}));
export var RotateEncryptionKeyResult;
(function (RotateEncryptionKeyResult) {
    RotateEncryptionKeyResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RotateEncryptionKeyResult || (RotateEncryptionKeyResult = {}));
