// @aws-sdk/client-redshift-data/dist-es/commands/DescribeStatementCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeStatementRequestFilterSensitiveLog, DescribeStatementResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_json1_1DescribeStatementCommand, serializeAws_json1_1DescribeStatementCommand, } from "../protocols/Aws_json1_1";
var DescribeStatementCommand = (function (_super) {
    __extends(DescribeStatementCommand, _super);
    function DescribeStatementCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeStatementCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftDataClient";
        var commandName = "DescribeStatementCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeStatementRequestFilterSensitiveLog,
            outputFilterSensitiveLog: DescribeStatementResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeStatementCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1DescribeStatementCommand(input, context);
    };
    DescribeStatementCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1DescribeStatementCommand(output, context);
    };
    return DescribeStatementCommand;
}($Command));
export { DescribeStatementCommand };
