// @aws-sdk/client-redshift-data/dist-es/commands/GetStatementResultCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetStatementResultRequestFilterSensitiveLog, GetStatementResultResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_json1_1GetStatementResultCommand, serializeAws_json1_1GetStatementResultCommand, } from "../protocols/Aws_json1_1";
var GetStatementResultCommand = (function (_super) {
    __extends(GetStatementResultCommand, _super);
    function GetStatementResultCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetStatementResultCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftDataClient";
        var commandName = "GetStatementResultCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetStatementResultRequestFilterSensitiveLog,
            outputFilterSensitiveLog: GetStatementResultResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetStatementResultCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1GetStatementResultCommand(input, context);
    };
    GetStatementResultCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1GetStatementResultCommand(output, context);
    };
    return GetStatementResultCommand;
}($Command));
export { GetStatementResultCommand };
