// @aws-sdk/client-redshift-data/dist-es/commands/ListSchemasCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListSchemasRequestFilterSensitiveLog, ListSchemasResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_json1_1ListSchemasCommand, serializeAws_json1_1ListSchemasCommand, } from "../protocols/Aws_json1_1";
var ListSchemasCommand = (function (_super) {
    __extends(ListSchemasCommand, _super);
    function ListSchemasCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListSchemasCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftDataClient";
        var commandName = "ListSchemasCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListSchemasRequestFilterSensitiveLog,
            outputFilterSensitiveLog: ListSchemasResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListSchemasCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1ListSchemasCommand(input, context);
    };
    ListSchemasCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1ListSchemasCommand(output, context);
    };
    return ListSchemasCommand;
}($Command));
export { ListSchemasCommand };
